lib/puppet/resource/catalog.rb

```diff
@@ -195,6 +195,14 @@ def alias(resource, key)
   def apply(options = {})
     Puppet::Util::Storage.load if host_config?
 
+    # The only option still supported is report.
+    # Issue a deprecation warning for any others.
+    deprecated_options = options.keys - [:report]
+    if !deprecated_options.empty?
+      Puppet.deprecation_warning("Puppet::Resource::Catalog.apply no longer " \
+                                 "takes the #{deprecated_options} option(s)")
```

Comment:
It looks like several applications have `options` that get passed blindly to the configurer and then to this method, e.g. `:pluginsync`, `:network_device`, ...